### PR TITLE
Made changes in the ParseOptionTickerOSI function. Now it can handle …

### PR DESCRIPTION
**Pull request by keshavdalmia10**
…both with and without space

#### Description

GenerateOptionTickerOSI/ParseOptionTickerOSI Improvement #8205

#### Types of changes

- [] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/SymbolRepresentation.cs [hunk 1]
```diff
@@ -338,40 +339,39 @@ public static Symbol ParseOptionTickerOSI(string ticker, SecurityType securityTy
...
-                right = OptionRight.Call;
-            }
-            else if (ticker[12] == 'P' || ticker[12] == 'p')
-            {
-                right = OptionRight.Put;
-            }
-            else
+            
+            var match = Regex.Match(ticker, @"^([A-Z]+)\s*(\d{6})([CP])(\d{8})$", RegexOptions.IgnoreCase);
+            if (!match.Success)
```
Comment (Martin-Molinero) [inline Common/SymbolRepresentation.cs:344]: Let's define this as a private readonly static field at the top, compiled regex, so we reuse it 👍

File: Common/SymbolRepresentation.cs [hunk 2]
```diff
@@ -338,40 +339,39 @@ public static Symbol ParseOptionTickerOSI(string ticker, SecurityType securityTy
...
             else
             {
                 throw new NotImplementedException($"ParseOptionTickerOSI(): {Messages.SymbolRepresentation.SecurityTypeNotImplemented(securityType)}");
             }
+
             var sid = SecurityIdentifier.GenerateOption(expiration, underlyingSid, optionTicker, market, strike, right, optionStyle);
             return new Symbol(sid, ticker, new Symbol(underlyingSid, underlyingSid.Symbol));
         }
+    
+
```
Comment (Martin-Molinero) [inline Common/SymbolRepresentation.cs:374]: Let's also add a new method `GenerateOptionTickerOSICompact` with this new format without spaces and add unit tests for it 👍 

nit - let's revert unrequired space changes like these

File: Tests/Common/SymbolRepresentationTests.cs [hunk 3]
```diff
@@ -34,6 +34,9 @@ public void OptionSymbolAliasMatchesOSI()
 
         [TestCase("SPXW  230111C02400000", SecurityType.IndexOption, OptionStyle.European, "SPXW", "SPX")]
         [TestCase("SPY   230111C02400000", SecurityType.Option, OptionStyle.American, "SPY", "SPY")]
+        [TestCase("AAPL240614C00100000", SecurityType.Option, OptionStyle.American, "AAPL", "AAPL")]
+        [TestCase("MSFT240614C00150000", SecurityType.Option, OptionStyle.American, "MSFT", "MSFT")]
+
```
Comment (Martin-Molinero) [inline Tests/Common/SymbolRepresentationTests.cs:39]: Nice! Thank you, let's please expand the unit test to actually assert the Option Strike, Style and Expiration date too, can add these as testcase arguments

**Review by Martin-Molinero — CHANGES_REQUESTED** [review]
Hey @keshavdalmia10! Welcome to Lean, thanks for the contribution. Leaving some requests

File: Common/SymbolRepresentation.cs (hunk 1 above)
Comment (keshavdalmia10) [reply Common/SymbolRepresentation.cs:344]: sure will work onit

File: Common/SymbolRepresentation.cs (hunk 2 above)
Comment (keshavdalmia10) [reply Common/SymbolRepresentation.cs:374]: sure will work on it

File: Tests/Common/SymbolRepresentationTests.cs (hunk 3 above)
Comment (keshavdalmia10) [reply Tests/Common/SymbolRepresentationTests.cs:39]: will do the necessary changes as you  told

**Pull request closed without merging**